core/src/test/java/org/elasticsearch/index/percolator/PercolatorQueryCacheTests.java

```diff
@@ -105,9 +103,12 @@ void initialize(Object... fields) throws IOException {
                 .build();
 
         Map<String, Tuple<ParseField, QueryParser<?>>> queryParsers = new HashMap<>();
-        queryParsers.put(TermQueryBuilder.NAME, new Tuple<>(TermQueryParser.QUERY_NAME_FIELD, new TermQueryParser()));
+        // It isn't clear to me why the casts are needed but ok
```

Comment:
this is funny indeed, the following in two lines doesn't require casting:
```
QueryParser<TermQueryBuilder> queryParser = TermQueryBuilder::fromXContent;
        queryParsers.put(TermQueryBuilder.NAME, new Tuple<>(TermQueryBuilder.QUERY_NAME_FIELD, queryParser));
```

settle for one of the variants and remove the comment?